src/main.rs

```diff
@@ -17,32 +18,41 @@ impl WeatherResponse {
         let url = Url::parse(&*url)?;
         println!("{}", url);
         let res = reqwest::get(url).await?.json::<serde_json::Value>().await?;
-        //unwrap current field of res or handle error   
-        match res.get("current") {
-            Some(x) => {
-                match _x.get("temp_c") {  
-                    Some(y) => {
-                        println!("current field found");
-                        let weather_res = WeatherResponse {temp_c: y.as_f64().unwrap() as f32};
-                        Ok(weather_res)
-                    },
-                    None => {
-                        println!("temp_c field not found");
-                        let weather_res = WeatherResponse {temp_c: 0.0};
-                        Ok(weather_res)
-                    }
+        
+        let temp = res.get("current").and_then(|x| x.get("temp_c")).and_then(|x| x.as_f64());
+        match temp {
+            Some(x) => Ok(WeatherResponse{temp_c: x as f32, forecast: None}),
+            None => Err(ExitFailure::from(std::io::Error::new(std::io::ErrorKind::Other, "API Error")))
+        }
+    }
+    async fn get_today(location: &String, api_key: &String) -> Result<Self, ExitFailure> {
+        let url = format!(
+            "http://api.weatherapi.com/v1/forecast.json?key={}&q={}&days=1&aqi=no&alerts=no",
+            location, api_key);
 
+        let url = Url::parse(&*url)?;
+        // println!("{}", url);
+        let res = reqwest::get(url).await?.json::<serde_json::Value>().await?;
+        let forcastday = res.get("forecast").and_then(|x| x.get("forecastday"));
```

Comment:
I think that now that you know the structure for the returned value, you should try and create a struct for it 
and use. srede::Deserialize attribute to make it parseable. If you want a hint for how the struct (and inner structs) should look like, you can ask me, but I think this is a good exercise.


parsing this complex value on your own is a bit combersome.